lib/CGameState.cpp

```diff
@@ -1866,8 +1873,8 @@ void CGameState::initVisitingAndGarrisonedHeroes()
 		{
 			for(CGTownInstance *t : k->second.towns)
 			{
-				int3 vistile = t->pos; vistile.x--; //tile next to the entrance
-				if(vistile == h->pos || h->pos==t->pos)
+				int3 vistile = t->visitablePos(); vistile.x++; //tile next to the entrance
```

Comment:
Why we changed -- to ++? Also might be worth to actually take all tiles around visitablePos and test them for accessibility and put hero on the first accessible tile. This will be the safest way as for me

Reply:
It's a bugfix, don't want to take extra scope here.

Before it was townpos.x - 2. Hero pos was townpos.x -2 + 1, which translated into townpos.x - 1
Now it's visitablePos, hero pos is visitablepos + 1

Why hero pos is x + 1? Don't know